types/bellatrix/bid.yaml

```
@@ -1,18 +1,24 @@
 Bellatrix:
-  BuilderBid:
+  BuilderBidCommon:
     type: object
-    description: "The `BuilderBid` object from the Builder API spec."
     properties:
-      header:
-        $ref: "../../beacon-apis/types/bellatrix/execution_payload.yaml#/Bellatrix/ExecutionPayloadHeader"
-        description: "`ExecutionPayloadHeader` to use in block proposal."
       value:
         $ref: "../../beacon-apis/types/primitive.yaml#/Uint256"
         description: "Payment in wei that will be paid to the `fee_recipient` account."
       pubkey:
         $ref: "../../beacon-apis/types/primitive.yaml#/Pubkey"
         description: "BLS public key of builder."
 
+  BuilderBid:
+    description: "The `BuilderBid` object from the Builder API Bellatrix spec."
+    allOf:
+      - $ref: '#/Bellatrix/BuilderBidCommon'
+      - type: object
+        properties:
+          header:
+            $ref: "../../beacon-apis/types/bellatrix/execution_payload.yaml#/Bellatrix/ExecutionPayloadHeader"
+            description: "`ExecutionPayloadHeader` to use in block proposal."
```

```suggestion
      - type: object
        properties:
          header:
            $ref: "../../beacon-apis/types/bellatrix/execution_payload.yaml#/Bellatrix/ExecutionPayloadHeader"
            description: "`ExecutionPayloadHeader` to use in block proposal."
      - $ref: '#/Bellatrix/BuilderBidCommon'
```

I think this is the backwards-compatible thing

---

Committed your suggestion, makes sense to me I think. For my understanding, so this just makes sure the "common" schema doesn't get overriden by the other right?

---

I was just trying to indicate the relative ordering of the accepted object

JSON objects don't have order within one "key scope" so its minor but I think what you had would put the header _after_ all of the "common" stuff and I believe the header was given first

I intend for my suggestion to put them in the same order